src/cudadev/CUDADataFormats/SiPixelDigisCUDA.h

```diff
@@ -6,10 +6,62 @@
 #include "CUDACore/cudaCompat.h"
 #include "CUDACore/device_unique_ptr.h"
 #include "CUDACore/host_unique_ptr.h"
+#include "DataFormats/SoAStore.h"
+#include "DataFormats/SoAView.h"
 
 class SiPixelDigisCUDA {
 public:
-  SiPixelDigisCUDA() = default;
+  generate_SoA_store(DeviceOnlyStore,
+    /* These are consumed by downstream device code                                                   */
+    SoA_column(uint16_t, xx),         /* local coordinates of each pixel                              */
+    SoA_column(uint16_t, yy),         /*                                                              */
+    SoA_column(uint16_t, moduleInd)   /* module id of each pixel                                      */
+  );
+  
+  generate_SoA_store(HostDeviceStore,
+    /* These are also transferred to host (see HostDataView) */
+    SoA_column(uint16_t, adc),        /* ADC of each pixel                                            */
+    SoA_column(int32_t, clus),        /* cluster id of each pixel                                     */
+    /* These are for CPU output; should we (eventually) place them to a                               */
+    /* separate product?                                                                              */
+    SoA_column(uint32_t, pdigi),     /* packed digi (row, col, adc) of each pixel                     */
+    SoA_column(uint32_t, rawIdArr)   /* DetId of each pixel                                           */
+  );
+  
+  generate_SoA_view(DeviceFullView,
+    SoA_view_store_list(
+      SoA_view_store(DeviceOnlyStore, deviceOnly),
+      SoA_view_store(HostDeviceStore, hostDevice)
+    ),
+    SoA_view_value_list(
+      SoA_view_value(deviceOnly, xx, xx),    /* local coordinates of each pixel                       */
+      SoA_view_value(deviceOnly, yy, yy),    /*                                                       */
+      SoA_view_value(deviceOnly, moduleInd, moduleInd),  /* module id of each pixel                   */
+      SoA_view_value(hostDevice, adc, adc),  /* ADC of each pixel                                     */
+      SoA_view_value(hostDevice, clus, clus),/* cluster id of each pixel                              */
+      SoA_view_value(hostDevice, pdigi, pdigi), /* packed digi (row, col, adc) of each pixel          */
+      SoA_view_value(hostDevice, rawIdArr, rawIdArr)  /* DetId of each pixel                          */
+      /* TODO: simple, no rename interface */
+    )    
+  );
+
+  /* Device pixel view: this is a second generation view (view from view) */
+  generate_SoA_const_view(DevicePixelView,
```

Comment:
Does the `DevicePixelView` have `const`-only interface? Including `Const` somewhere in the class name would make that clear in places where the class name is used.

Reply:
Sure. I did it,